game/test/ecs/systems/XPSystemTest.java

```diff
@@ -0,0 +1,73 @@
+package ecs.systems;
+
+import static org.junit.Assert.*;
+
+import ecs.components.xp.ILevelUp;
+import ecs.components.xp.XPComponent;
+import ecs.entities.Entity;
+import mydungeon.ECS;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class XPSystemTest {
+
+    private Entity entity;
+    private XPComponent xpComponent;
+    private XPSystem xpSystem;
+    private ILevelUp levelUp;
+
+    @Before
+    public void setup() {
+        ECS.entities.clear();
+        ECS.systems = new SystemController();
+        this.entity = new Entity();
+        this.levelUp = Mockito.mock(ILevelUp.class);
+        this.xpComponent = new XPComponent(entity, this.levelUp);
+        this.xpSystem = new XPSystem();
```

Comment:
Entweder immer this oder nur this wenn es benötigt wird hier z.b. this.entity = new Entity(), mit einem this aber dann bei new XPComponent(entity, this.levelUp)

Ich würde es lieber immer weglassen anstelle von es immer hinzuschreiben.

Reply:
Stimmt, wenn dann konsistent durch. Ich bin dann aber tatsächlich dafür immer `this` zu schreiben, dann ist direkt klar, dass es sich um ein Attribut der Klasse handelt.